src/Analyzers/CSharp/Analyzers/NewLines/ConditionalExpressionPlacement/ConditionalExpressionPlacementDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.NewLines.ConditionalExpressionPlacement
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class ConditionalExpressionPlacementDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+        public ConditionalExpressionPlacementDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.ConditionalExpressionPlacementDiagnosticId,
+                   EnforceOnBuildValues.ConsecutiveBracePlacement,
+                   CSharpCodeStyleOptions.AllowBlankLineAfterTokenInConditionalExpression,
+                   new LocalizableResourceString(
+                       nameof(CSharpAnalyzersResources.Blank_line_not_allowed_after_conditional_expression_token), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SyntaxTreeWithoutSemanticsAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterSyntaxTreeAction(AnalyzeTree);
+
+        private void AnalyzeTree(SyntaxTreeAnalysisContext context)
+        {
+            var option = context.GetCSharpAnalyzerOptions().AllowBlankLineAfterTokenInConditionalExpression;
+            if (option.Value)
+                return;
```

Comment:
I'd love if the option can let the user choose between enforcing one of the two styles

```csharp
_ = expression ?
    when_true :
    when_false;
```

```csharp
_ = expression
    ? when_true
    : when_false;
```

Currently the option either disables the analysis or enforces the latter.

cc @mavasani for thoughts as well.

Reply:
Won't fix for this.  That's not what these experimental options do :)

Reply:
I agree with @Youssef1313 - I have seen users who prefer enforcing the former style, so if there is anyway we can get a 3-state option value (enforce style a, or enforce style b or disable), then that would be fantastic.

1. Style a: Multiline conditional expression must have a newline following the `?` and `:` tokens
2. Style b: Multiline conditional expression must not have a newline following the `?` or `:` tokens

Reply:
ok.  i'm not opposed to that if someone wants to add it.   :)

Reply:
Filed #65479